Tarefa de Análise do Processo Seletivo UFRJ Analytica

Por: Juliana Barros

A permanência e a qualidade dos estudos de alunos do ensino médio das escolas da Região Metropolitana

Para a análise, eu escolhi estudar quais aspectos apresentam relação na qualidade dos estudos e permanência dos alunos nas escolas da região metropolitana do Rio de Janeiro. Assim, montei algumas hipóteses e busquei relacioná-las com as taxas de abandono.

A escolha da região foi devido ao fato dela conter quase 75% da população do estado do Rio, segundo o censo do IBGE de 2010 [1]. A Região Metropolitana é formada por 19 municípios: 
Belford Roxo, Duque de Caxias, Guapimirim, Itaboraí, Itaguaí, Japeri, Magé, Maricá, Mesquita, Nilópolis, Niterói, Nova Iguaçu, Paracambi, Queimados, Rio de Janeiro, São Gonçalo, São João de Meriti, Seropédica e Tanguá.
Maparj2018RMRJ-Mapa_Região_Metropolitana_do_Rio_de_Janeiro_Brasil-2018.png

No site do IBGE, encontrei os códigos dos municípios, e coloquei em uma lista em ordem alfabética [2].

In [ ]:
id_municipios = ['3300456', '3301702', '3301850', '3301900', '3302007', '3302270', '3302502', '3302700', '3302858',
                 '3303203', '3303302', '3303500', '3303609', '3304144', '3304557', '3304904', '3305109', '3305554', '3305752']

Exportei do BigQuery do Google Cloud, as colunas de interesse de todas as escolas de cada município, fiz um dataframe para cada município e um de toda a região metropolitana para avaliar quais indicadores poderiam ser usados.

In [3]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

In [11]:
#Belford Roxo - 3300456
belford_roxo = pd.read_csv('belford_roxo.csv')
# Duque de Caxias - 3301702
duque_caxias =  pd.read_csv('duque_caxias.csv')
# Guapimirim - 3301850
guapimirim = pd.read_csv('guapimirim.csv')
# Itaboraí - 3301900
itaborai = pd.read_csv('itaborai.csv')
# Itaguaí - 3302007
itaguai = pd.read_csv('itaguai.csv')
# Japeri - 3302270
japeri = pd.read_csv('japeri.csv')
# Magé - 3302502
mage = pd.read_csv('mage.csv')
# Maricá - 3302700
marica = pd.read_csv('marica.csv')
# Mesquita - 3302858
mesquita = pd.read_csv('mesquita.csv')
# Nilópolis - 3303203
nilopolis = pd.read_csv('nilopolis.csv')
# Niterói - 3303302
niteroi = pd.read_csv('niteroi.csv')
# Nova Iguaçu- 3303500
nova_iguacu = pd.read_csv('nova_iguacu.csv')
# Paracambi - 3303609
paracambi = pd.read_csv('paracambi.csv')
# Queimados - 3304144
queimados = pd.read_csv('queimados.csv')
# Rio de Janeiro - 3304557
rio_de_janeiro = pd.read_csv('rio_de_janeiro.csv')
# São Gonçalo - 3304904
sao_goncalo = pd.read_csv('sao_goncalo.csv')
# São João de Meriti - 3305109
sao_joao_de_meriti = pd.read_csv('sao_joao_de_meriti.csv')
# Seropédica - 3305554
seropedica = pd.read_csv('seropedica.csv')
# Tanguá - 3305752
tangua = pd.read_csv('tangua.csv')

municipios = [belford_roxo, duque_caxias, guapimirim, itaborai, itaguai, japeri, mage, marica, mesquita, nilopolis,
            niteroi, nova_iguacu, paracambi, queimados, rio_de_janeiro, sao_goncalo, sao_joao_de_meriti, seropedica, tangua]

regiao = pd.concat(municipios)
regiao.shape[0]

69349

Para facilitar a visualização, troquei o nome das colunas dos dataframes.

In [12]:
dict = {'atu_em': 'media_alunos_turma_em',
        'atu_em_1_ano': 'media_alunos_turma_1',
        'atu_em_2_ano': 'media_alunos_turma_2',
        'atu_em_3_ano': 'media_alunos_turma_3',
        'atu_em_4_ano': 'media_alunos_turma_4',
        'dsu_em': 'prof_ensino_superior',
        'tdi_em': 'distocao_idade_em',
        'tdi_em_1_ano': 'distocao_idade_1',
        'tdi_em_2_ano': 'distocao_idade_2',
        'tdi_em_3_ano': 'distocao_idade_3',
        'tdi_em_4_ano': 'distocao_idade_4',
        'had_em': 'media_hora_em',
        'had_em_1_ano' : 'media_hora_1',
        'had_em_2_ano' : 'media_hora_2',
        'had_em_3_ano' : 'media_hora_3',
        'had_em_4_ano' : 'media_hora_4',
        }
 
regiao.rename(columns=dict,
          inplace=True)
for municipio in municipios:
        municipio.rename(columns=dict, inplace = True)

Algumas informações básicas sobre o dataset total:

In [7]:
regiao

id_municipio  id_escola localizacao      rede  media_alunos_turma_em  \
0         3300456   33046247      urbana  estadual                   37.9   
1         3300456   33046263      urbana  estadual                   31.2   
2         3300456   33046271      urbana  estadual                    NaN   
3         3300456   33046280      urbana  estadual                    NaN   
4         3300456   33046298      urbana  estadual                    NaN   
..            ...        ...         ...       ...                    ...   
332       3305752   33099944      urbana   privada                    NaN   
333       3305752   33099952      urbana   privada                    NaN   
334       3305752   33154430      urbana   privada                    NaN   
335       3305752   33162417      urbana   privada                    NaN   
336       3305752   33329206      urbana   privada                    NaN   

     media_alunos_turma_1  media_alunos_turma_2  media_alunos_turma_3  \
0                    43.0                  29.2                  43.5   
1                    36.0                  27.8                  32.3   
2                     NaN                   NaN                   NaN   
3                     NaN                   NaN                   NaN   
4                     NaN                   NaN                   NaN   
..                    ...                   ...                   ...   
332                   NaN                   NaN                   NaN   
333                   NaN                   NaN                   NaN   
334                   NaN                   NaN                   NaN   
335                   NaN                   NaN                   NaN   
336                   NaN                   NaN                   NaN   

     media_alunos_turma_4  afd_em_grupo_1  ...  distocao_idade_1  \
0                     NaN             NaN  ...              60.9   
1                     NaN             NaN  ...              41.7   
2                     NaN             NaN  ...               NaN   
3                     NaN             NaN  ...               NaN   
4                     NaN             NaN  ...               NaN   
..                    ...             ...  ...               ...   
332                   NaN             NaN  ...               NaN   
333                   NaN             NaN  ...               NaN   
334                   NaN             NaN  ...               NaN   
335                   NaN             NaN  ...               NaN   
336                   NaN             NaN  ...               NaN   

     distocao_idade_2  distocao_idade_3  distocao_idade_4  taxa_abandono_em  \
0                60.0              56.9               NaN              13.8   
1                64.0              78.3               NaN              12.3   
2                 NaN               NaN               NaN               NaN   
3                 NaN               NaN               NaN               NaN   
4                 NaN               NaN               NaN               NaN   
..                ...               ...               ...               ...   
332               NaN               NaN               NaN               NaN   
333               NaN               NaN               NaN               NaN   
334               NaN               NaN               NaN               NaN   
335               NaN               NaN               NaN               NaN   
336               NaN               NaN               NaN               NaN   

     taxa_abandono_em_1_ano  taxa_abandono_em_2_ano  taxa_abandono_em_3_ano  \
0                      17.5                    14.7                     7.6   
1                       2.8                    18.3                    12.5   
2                       NaN                     NaN                     NaN   
3                       NaN                     NaN                     NaN   
4                       NaN               

In [8]:
regiao.describe()

id_municipio     id_escola  media_alunos_turma_em  \
count  6.934900e+04  6.934900e+04           15245.000000   
mean   3.303665e+06  3.312108e+07              28.832122   
std    1.311725e+03  8.347347e+04              10.520464   
min    3.300456e+06  3.303887e+07               1.000000   
25%    3.302700e+06  3.306838e+07              22.800000   
50%    3.304557e+06  3.310199e+07              29.500000   
75%    3.304557e+06  3.314748e+07              35.300000   
max    3.305752e+06  3.355823e+07             135.300000   

       media_alunos_turma_1  media_alunos_turma_2  media_alunos_turma_3  \
count          14846.000000          14447.000000          14042.000000   
mean              31.267803             28.754170             28.229775   
std               11.646656             11.353806             11.811110   
min                1.000000              1.000000              1.000000   
25%               25.000000             21.800000             20.500000   
50%               31.900000             29.000000             28.500000   
75%               38.000000             35.900000             35.500000   
max              165.000000            141.000000            148.000000   

       media_alunos_turma_4  afd_em_grupo_1  media_hora_em  media_hora_1  ...  \
count            301.000000     7898.000000   11426.000000  11140.000000  ...   
mean              30.998339       67.626982       5.447873      5.501544  ...   
std               12.047012       15.878741       0.966853      1.052241  ...   
min                1.000000        0.000000       2.900000      3.000000  ...   
25%               24.000000       59.200000       5.000000      5.100000  ...   
50%               32.300000       69.600000       5.300000      5.300000  ...   
75%               39.000000       78.600000       5.500000      5.500000  ...   
max               88.000000      100.000000      13.700000     15.800000  ...   

       distocao_idade_1  distocao_idade_2  distocao_idade_3  distocao_idade_4  \
count      14609.000000      14060.000000      13462.000000        292.000000   
mean          37.001499         33.738215         30.956537         42.062671   
std           24.938634         24.667209         24.878556         24.389925   
min            0.000000          0.000000          0.000000          0.000000   
25%           14.400000         12.500000         10.000000         24.850000   
50%           34.500000         29.700000         25.000000         35.750000   
75%           54.900000         50.000000         46.100000         53.225000   
max          100.000000        100.000000        100.000000        100.000000   

       taxa_abandono_em  taxa_abandono_em_1_ano  taxa_abandono_em_2_ano  \
count      12562.000000            12252.000000            11936.000000   
mean           7.436666                9.047478                7.323400   
std           10.389449               12.617983               10.641663   
min            0.000000                0.000000                0.000000   
25%            0.000000                0.000000                0.000000   
50%            2.700000                3.400000                2.200000   
75%           11.200000               13.800000               11.100000   
max          100.000000              100.000000              100.000000   

       taxa_abandono_em_3_ano  taxa_abandono_em_4_ano  prof_ensino_superior  
count            11624.000000              284.000000          10043.000000  
mean                 5.201385                4.139789             94.814498  
std                  8.204316                5.869355              8.789687  
min                  0.000000                0.000000              0.000000  
25%                  0.000000                0.000000             92.900000  
50%                  1.000000                2.200000            100.000000  
75%                  7.700000                5.225000            100.000000  
max                100.000000 

Algumas das escolas possuem 4 anos de ensino médio devido ao ensino técnico. Como, no máximo, apenas 301 das 69344 escolas (0.4%) ofereceram dados sobre algum dos índices para o quarto ano do ensino médio, e esse tipo de ensino não é o foco da análise, retirei essas escolas.

In [85]:
#quantidade de escolas que responderam a cada um dos indices
df = regiao[regiao['media_alunos_turma_4'].notna()]
print(df.shape[0])

df = regiao[regiao['distocao_idade_4'].notna()]
print(df.shape[0])

df = regiao[regiao['taxa_abandono_em_4_ano'].notna()]
print(df.shape[0])

df = regiao[regiao['media_hora_4'].notna()]
print(df.shape[0])


301
292
284
169


In [13]:
tecnico = ['media_alunos_turma_4', 'distocao_idade_4', 'taxa_abandono_em_4_ano', 'media_hora_4']

for indice in tecnico:
    regiao = regiao[regiao[indice].isna()]
    regiao = regiao.drop(indice, axis=1)

regiao

id_municipio  id_escola localizacao      rede  media_alunos_turma_em  \
0         3300456   33046247      urbana  estadual                   37.9   
1         3300456   33046263      urbana  estadual                   31.2   
2         3300456   33046271      urbana  estadual                    NaN   
3         3300456   33046280      urbana  estadual                    NaN   
4         3300456   33046298      urbana  estadual                    NaN   
..            ...        ...         ...       ...                    ...   
332       3305752   33099944      urbana   privada                    NaN   
333       3305752   33099952      urbana   privada                    NaN   
334       3305752   33154430      urbana   privada                    NaN   
335       3305752   33162417      urbana   privada                    NaN   
336       3305752   33329206      urbana   privada                    NaN   

     media_alunos_turma_1  media_alunos_turma_2  media_alunos_turma_3  \
0                    43.0                  29.2                  43.5   
1                    36.0                  27.8                  32.3   
2                     NaN                   NaN                   NaN   
3                     NaN                   NaN                   NaN   
4                     NaN                   NaN                   NaN   
..                    ...                   ...                   ...   
332                   NaN                   NaN                   NaN   
333                   NaN                   NaN                   NaN   
334                   NaN                   NaN                   NaN   
335                   NaN                   NaN                   NaN   
336                   NaN                   NaN                   NaN   

     afd_em_grupo_1  media_hora_em  ...  media_hora_3  distocao_idade_em  \
0               NaN            NaN  ...           NaN               59.5   
1               NaN            NaN  ...           NaN               64.7   
2               NaN            NaN  ...           NaN                NaN   
3               NaN            NaN  ...           NaN                NaN   
4               NaN            NaN  ...           NaN                NaN   
..              ...            ...  ...           ...                ...   
332             NaN            NaN  ...           NaN                NaN   
333             NaN            NaN  ...           NaN                NaN   
334             NaN            NaN  ...           NaN                NaN   
335             NaN            NaN  ...           NaN                NaN   
336             NaN            NaN  ...           NaN                NaN   

     distocao_idade_1  distocao_idade_2  distocao_idade_3  taxa_abandono_em  \
0                60.9              60.0              56.9              13.8   
1                41.7              64.0              78.3              12.3   
2                 NaN               NaN               NaN               NaN   
3                 NaN               NaN               NaN               NaN   
4                 NaN               NaN               NaN               NaN   
..                ...               ...               ...               ...   
332               NaN               NaN               NaN               NaN   
333               NaN               NaN               NaN               NaN   
334               NaN               NaN               NaN               NaN   
335               NaN               NaN               NaN               NaN   
336               NaN               NaN               NaN               NaN   

     taxa_abandono_em_1_ano  taxa_abandono_em_2_ano  taxa_abandono_em_3_ano  \
0                      17.5                    14.7                     7.6   
1                       2.8                    18.3                    12.5   
2                       NaN                     NaN                     NaN   
3                       N

In [32]:
regiao.groupby('id_municipio')[['id_escola', 'localizacao', 'rede',
       'media_alunos_turma_em', 'media_alunos_turma_1', 'media_alunos_turma_2',
       'media_alunos_turma_3', 'afd_em_grupo_1', 'media_hora_em',
       'media_hora_1', 'media_hora_2', 'media_hora_3', 'distocao_idade_em',
       'distocao_idade_1', 'distocao_idade_2', 'distocao_idade_3',
       'taxa_abandono_em', 'taxa_abandono_em_1_ano', 'taxa_abandono_em_2_ano',
       'taxa_abandono_em_3_ano', 'prof_ensino_superior']].apply(lambda x: x.isna().mean())

id_escola  localizacao  rede  media_alunos_turma_em  \
id_municipio                                                        
3300456             0.0          0.0   0.0               0.784221   
3301702             0.0          0.0   0.0               0.768679   
3301850             0.0          0.0   0.0               0.907692   
3301900             0.0          0.0   0.0               0.815451   
3302007             0.0          0.0   0.0               0.813467   
3302270             0.0          0.0   0.0               0.813765   
3302502             0.0          0.0   0.0               0.817468   
3302700             0.0          0.0   0.0               0.796748   
3302858             0.0          0.0   0.0               0.843038   
3303203             0.0          0.0   0.0               0.724286   
3303302             0.0          0.0   0.0               0.747770   
3303500             0.0          0.0   0.0               0.739412   
3303609             0.0          0.0   0.0               0.807860   
3304144             0.0          0.0   0.0               0.753672   
3304557             0.0          0.0   0.0               0.798205   
3304904             0.0          0.0   0.0               0.752702   
3305109             0.0          0.0   0.0               0.796032   
3305554             0.0          0.0   0.0               0.787709   
3305752             0.0          0.0   0.0               0.916914   

              media_alunos_turma_1  media_alunos_turma_2  \
id_municipio                                               
3300456                   0.787706              0.793093   
3301702                   0.778194              0.787085   
3301850                   0.907692              0.912821   
3301900                   0.817358              0.823557   
3302007                   0.814241              0.815015   
3302270                   0.819163              0.823212   
3302502                   0.824828              0.828263   
3302700                   0.798374              0.802439   
3302858                   0.843038              0.847257   
3303203                   0.740714              0.746429   
3303302                   0.754615              0.759593   
3303500                   0.746388              0.754194   
3303609                   0.838428              0.840611   
3304144                   0.753672              0.761582   
3304557                   0.803434              0.807883   
3304904                   0.756198              0.764940   
3305109                   0.799944              0.808606   
3305554                   0.789944              0.792179   
3305752                   0.916914              0.916914   

              media_alunos_turma_3  afd_em_grupo_1  media_hora_em  \
id_municipio                                                        
3300456                   0.802915        0.867554       0.822877   
3301702                   0.795352        0.853533       0.808142   
3301850                   0.912821        0.945299       0.924786   
3301900                   0.832141        0.886505       0.850262   
3302007                   0.819659        0.886997       0.848297   
3302270                   0.829960        0.886640       0.850202   
3302502                   0.834642        0.890579       0.852306   
3302700                   0.808943        0.875610       0.833333   
3302858                   0.848945        0.904641       0.873418   
3303203                   0.752857        0.843571       0.784286   
3303302                   0.765194        0.849201       0.797138   
3303500                   0.762830        0.831589       0.781930   
3303609                   0.847162        0.888646       0.847162   
3304144                   0.767232        0.845198       0.800000   
3304557                   0.811629        0.930224       0.873756   
3304904                   0.773522        0.840114       0.794024   
3305109                   0.813076        0.875384       0.83

In [35]:
df = regiao.groupby(['id_municipio', 'localizacao'])[['id_escola',
       'media_alunos_turma_em', 'media_alunos_turma_1', 'media_alunos_turma_2',
       'media_alunos_turma_3', 'afd_em_grupo_1', 'media_hora_em',
       'media_hora_1', 'media_hora_2', 'media_hora_3', 'distocao_idade_em',
       'distocao_idade_1', 'distocao_idade_2', 'distocao_idade_3',
       'taxa_abandono_em', 'taxa_abandono_em_1_ano', 'taxa_abandono_em_2_ano',
       'taxa_abandono_em_3_ano', 'prof_ensino_superior']].apply(lambda x: x.isna().mean())
pd.set_option('display.max_rows', df.shape[0]+1)
df

id_escola  media_alunos_turma_em  \
id_municipio localizacao                                     
3300456      rural              0.0               1.000000   
             urbana             0.0               0.784015   
3301702      rural              0.0               0.800738   
             urbana             0.0               0.767264   
3301850      rural              0.0               0.696970   
             urbana             0.0               0.920290   
3301900      rural              0.0               0.896213   
             urbana             0.0               0.773844   
3302007      rural              0.0               0.859649   
             urbana             0.0               0.803571   
3302270      rural              0.0               1.000000   
             urbana             0.0               0.778846   
3302502      rural              0.0               0.878261   
             urbana             0.0               0.805080   
3302700      rural              0.0               0.962209   
             urbana             0.0               0.732506   
3302858      urbana             0.0               0.843038   
3303203      urbana             0.0               0.724286   
3303302      rural              0.0               1.000000   
             urbana             0.0               0.747613   
3303500      rural              0.0               0.879518   
             urbana             0.0               0.733368   
3303609      rural              0.0               1.000000   
             urbana             0.0               0.745665   
3304144      rural              0.0               0.767857   
             urbana             0.0               0.752714   
3304557      rural              0.0               0.880000   
             urbana             0.0               0.798125   
3304904      rural              0.0               0.878261   
             urbana             0.0               0.747938   
3305109      rural              0.0               1.000000   
             urbana             0.0               0.795918   
3305554      rural              0.0               0.811570   
             urbana             0.0               0.737931   
3305752      rural              0.0               1.000000   
             urbana             0.0               0.882845   

                          media_alunos_turma_1  media_alunos_turma_2  \
id_municipio localizacao                                               
3300456      rural                    1.000000              1.000000   
             urbana                   0.787504              0.792896   
3301702      rural                    0.800738              0.800738   
             urbana                   0.777199              0.786482   
3301850      rural                    0.696970              0.727273   
             urbana                   0.920290              0.923913   
3301900      rural                    0.896213              0.900421   
             urbana                   0.776734              0.783960   
3302007      rural                    0.859649              0.859649   
             urbana                   0.804511              0.805451   
3302270      rural                    1.000000              1.000000   
             urbana                   0.785256              0.790064   
3302502      rural                    0.881159              0.881159   
             urbana                   0.813349              0.817484   
3302700      rural                    0.962209              0.962209   
             urbana                   0.734763              0.740406   
3302858      urbana                   0.843038              0.847257   
3303203      urbana                   0.740714              0.746429   
3303302      rural                    1.000000              1.000000   
             urbana                   0.754462              0.759444   
3303500      rural                    0.879518              0.887550   
             

In [1]:
import plotly.express as px
px.box( df, x=' nome da coluna de média de nulo', color= 'tipo de escola')

ModuleNotFoundError: No module named 'plotly'

In [34]:

df = regiao.groupby(['id_municipio', 'rede'])[['id_escola', 'localizacao',
       'media_alunos_turma_em', 'media_alunos_turma_1', 'media_alunos_turma_2',
       'media_alunos_turma_3', 'afd_em_grupo_1', 'media_hora_em',
       'media_hora_1', 'media_hora_2', 'media_hora_3', 'distocao_idade_em',
       'distocao_idade_1', 'distocao_idade_2', 'distocao_idade_3',
       'taxa_abandono_em', 'taxa_abandono_em_1_ano', 'taxa_abandono_em_2_ano',
       'taxa_abandono_em_3_ano', 'prof_ensino_superior']].apply(lambda x: x.isna().mean())
pd.set_option('display.max_rows', df.shape[0]+1)
df

id_escola  localizacao  media_alunos_turma_em  \
id_municipio rede                                                       
3300456      estadual         0.0          0.0               0.189280   
             federal          0.0          0.0               1.000000   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.873637   
3301702      estadual         0.0          0.0               0.217722   
             federal          0.0          0.0               0.000000   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.809882   
3301850      estadual         0.0          0.0               0.047619   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.825000   
3301900      estadual         0.0          0.0               0.214286   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.729614   
3302007      estadual         0.0          0.0               0.080745   
             federal          0.0          0.0               0.571429   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.753425   
3302270      estadual         0.0          0.0               0.132353   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.872611   
3302502      estadual         0.0          0.0               0.044444   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.747788   
3302700      estadual         0.0          0.0               0.210145   
             federal          0.0          0.0               0.200000   
             municipal        0.0          0.0               0.992877   
             privada          0.0          0.0               0.657143   
3302858      estadual         0.0          0.0               0.193548   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.890681   
3303203      estadual         0.0          0.0               0.071823   
             federal          0.0          0.0               0.000000   
             municipal        0.0          0.0               0.996815   
             privada          0.0          0.0               0.772166   
3303302      estadual         0.0          0.0               0.340491   
             federal          0.0          0.0               0.161290   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.750574   
3303500      estadual         0.0          0.0               0.208297   
             federal          0.0          0.0               0.000000   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.790354   
3303609      estadual         0.0          0.0               0.122807   
             federal          0.0          0.0               0.000000   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.771429   
3304144      estadual         0.0          0.0               0.114286   
             municipal        0.0          0.0               1.000000   
             privada          0.0          0.0               0.783505   
3304557      estadual         0.0          0.0               0.145103   
             federal          0.0          0.0               0.466321   
             municipal        0.0          0.0               1.000000   
         

Para cada município, vamos analisar a quantidade de escolas que possuem poucos dados. 

In [28]:
#new = regiao(['id_municipio']=='3300456').copy()
new = regiao.loc[regiao['id_municipio'] == 3300456]
list = new.isnull().sum(axis=1).tolist()
i = 0
for n in list:
    if n>11:
        i+=1
print(i)
new.info()

2482
<class 'pandas.core.frame.DataFrame'>
Int64Index: 3156 entries, 0 to 3162
Data columns (total 22 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id_municipio            3156 non-null   int64  
 1   id_escola               3156 non-null   int64  
 2   localizacao             3156 non-null   object 
 3   rede                    3156 non-null   object 
 4   media_alunos_turma_em   681 non-null    float64
 5   media_alunos_turma_1    670 non-null    float64
 6   media_alunos_turma_2    653 non-null    float64
 7   media_alunos_turma_3    622 non-null    float64
 8   afd_em_grupo_1          418 non-null    float64
 9   media_hora_em           559 non-null    float64
 10  media_hora_1            549 non-null    float64
 11  media_hora_2            535 non-null    float64
 12  media_hora_3            509 non-null    float64
 13  distocao_idade_em       675 non-null    float64
 14  distocao_idade_1        665 non-nul

In [65]:
regiao.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 66306 entries, 0 to 336
Data columns (total 26 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id_escola               66306 non-null  int64  
 1   localizacao             66306 non-null  object 
 2   rede                    66306 non-null  object 
 3   media_alunos_turma_em   14418 non-null  float64
 4   media_alunos_turma_1    14027 non-null  float64
 5   media_alunos_turma_2    13689 non-null  float64
 6   media_alunos_turma_3    13311 non-null  float64
 7   afd_em_grupo_1          8611 non-null   float64
 8   media_hora_em           11287 non-null  float64
 9   media_hora_1            10994 non-null  float64
 10  media_hora_2            10742 non-null  float64
 11  media_hora_3            10420 non-null  float64
 12  distocao_idade_em       14118 non-null  float64
 13  distocao_idade_1        13826 non-null  float64
 14  distocao_idade_2        13358 non-null  

No banco de dados, alguns dos índices tinham dados tanto para o ensino médio como um todo, como para cada ano dele. Assim, importei todos e analisei se era viável usar o índice total, ou se era necessário fazer uma média dos índices dos três anos.

In [26]:
regiao

id_escola localizacao      rede  atu_em  atu_em_1_ano  atu_em_2_ano  \
0     33046247      urbana  estadual    37.9          43.0          29.2   
1     33046263      urbana  estadual    31.2          36.0          27.8   
2     33046271      urbana  estadual     NaN           NaN           NaN   
3     33046280      urbana  estadual     NaN           NaN           NaN   
4     33046298      urbana  estadual     NaN           NaN           NaN   
..         ...         ...       ...     ...           ...           ...   
332   33099944      urbana   privada     NaN           NaN           NaN   
333   33099952      urbana   privada     NaN           NaN           NaN   
334   33154430      urbana   privada     NaN           NaN           NaN   
335   33162417      urbana   privada     NaN           NaN           NaN   
336   33329206      urbana   privada     NaN           NaN           NaN   

     atu_em_3_ano  atu_em_4_ano  afd_em_grupo_1  had_em  ...  \
0            43.5           NaN             NaN     NaN  ...   
1            32.3           NaN             NaN     NaN  ...   
2             NaN           NaN             NaN     NaN  ...   
3             NaN           NaN             NaN     NaN  ...   
4             NaN           NaN             NaN     NaN  ...   
..            ...           ...             ...     ...  ...   
332           NaN           NaN             NaN     NaN  ...   
333           NaN           NaN             NaN     NaN  ...   
334           NaN           NaN             NaN     NaN  ...   
335           NaN           NaN             NaN     NaN  ...   
336           NaN           NaN             NaN     NaN  ...   

     taxa_abandono_em_1_ano  taxa_abandono_em_2_ano  taxa_abandono_em_3_ano  \
0                      17.5                    14.7                     7.6   
1                       2.8                    18.3                    12.5   
2                       NaN                     NaN                     NaN   
3                       NaN                     NaN                     NaN   
4                       NaN                     NaN                     NaN   
..                      ...                     ...                     ...   
332                     NaN                     NaN                     NaN   
333                     NaN                     NaN                     NaN   
334                     NaN                     NaN                     NaN   
335                     NaN                     NaN                     NaN   
336                     NaN                     NaN                     NaN   

     taxa_abandono_em_4_ano  tnr_em  tnr_em_1_ano  tnr_em_2_ano  tnr_em_3_ano  \
0                       NaN     NaN           NaN           NaN           NaN   
1                       NaN     NaN           NaN           NaN           NaN   
2                       NaN     NaN           NaN           NaN           NaN   
3                       NaN     NaN           NaN           NaN           NaN   
4                       NaN     NaN           NaN           NaN           NaN   
..                      ...     ...           ...           ...           ...   
332                     NaN     NaN           NaN           NaN           NaN   
333                     NaN     NaN           NaN           NaN           NaN   
334                     NaN     NaN           NaN           NaN           NaN   
335                     NaN     NaN           NaN           NaN           NaN   
336                     NaN     NaN           NaN           NaN           NaN   

     tnr_em_4_ano  dsu_em  
0             NaN     NaN  
1             NaN     NaN  
2             NaN     NaN  
3             NaN     NaN  
4             NaN     NaN  
..            ...     ...  
332           NaN     NaN  
333           NaN     NaN  
334           NaN     NaN  
335           NaN     NaN  
336           NaN     NaN  

[43652 rows x 30 columns]

In [37]:
df

id_escola localizacao      rede  atu_em  atu_em_1_ano  atu_em_2_ano  \
42     33047960      urbana  estadual    44.6          45.3          48.2   
221    33047960      urbana  estadual    37.1          36.2          36.6   
427    33047960      urbana  estadual    38.7          36.1          40.2   
874    33047960      urbana  estadual    43.5          43.2          45.9   
1062   33047960      urbana  estadual    39.1          39.8          38.0   
...         ...         ...       ...     ...           ...           ...   
235    33045496      urbana  estadual    29.4          30.6          31.7   
429    33045496      urbana  estadual    28.4          31.4          27.0   
582    33045496       rural  estadual    36.3          41.3          33.3   
645    33045496       rural  estadual    22.9          33.9          19.8   
774    33045496       rural  estadual    38.1          39.3          39.6   

      atu_em_3_ano  atu_em_4_ano  afd_em_grupo_1  had_em  ...  \
42            41.3          43.8             NaN     NaN  ...   
221           36.7          41.0             NaN     NaN  ...   
427           37.0          43.2             NaN     6.7  ...   
874           38.4          45.4             NaN     NaN  ...   
1062          36.5          42.3             NaN    11.3  ...   
...            ...           ...             ...     ...  ...   
235           26.8          30.7             NaN     7.1  ...   
429           26.7          30.0             NaN     6.7  ...   
582           32.1          37.2             NaN     NaN  ...   
645           19.5          19.1             NaN     6.0  ...   
774           32.9          41.0             NaN     NaN  ...   

      taxa_abandono_em_1_ano  taxa_abandono_em_2_ano  taxa_abandono_em_3_ano  \
42                      17.0                     9.6                     6.3   
221                     14.5                    12.5                    10.5   
427                     16.5                    18.8                    10.8   
874                     21.4                    15.3                     5.9   
1062                    11.0                     6.9                     2.7   
...                      ...                     ...                     ...   
235                      8.0                     4.9                     1.6   
429                     10.0                     5.7                     2.6   
582                     15.6                    16.3                    10.0   
645                      1.2                     2.7                     1.7   
774                      1.0                     0.0                     0.0   

      taxa_abandono_em_4_ano  tnr_em  tnr_em_1_ano  tnr_em_2_ano  \
42                       2.9     NaN           NaN           NaN   
221                      8.6     NaN           NaN           NaN   
427                      4.7     1.1           2.1           0.8   
874                     11.2     NaN           NaN           NaN   
1062                     7.1     1.7           3.0           2.1   
...                      ...     ...           ...           ...   
235                      1.1     3.4           6.6           2.1   
429                      1.1     4.9           6.5           5.9   
582                      9.2     NaN           NaN           NaN   
645                      3.0    13.9          17.2          19.1   
774                      0.0     NaN           NaN           NaN   

      tnr_em_3_ano  tnr_em_4_ano  dsu_em  
42             NaN           NaN     NaN  
221            NaN           NaN     NaN  
427            0.5           0.9     NaN  
874            NaN           NaN     NaN  
1062           0.7           0.6    93.8  
...            ...           ...     ...  
235            2.3           NaN    98.4  
429            4.4           NaN    97.1  
582            NaN           NaN     NaN  
645           11.3           0.7     NaN  
774            NaN           NaN     NaN  

[229 

In [29]:
colunas = regiao.columns
print(colunas)
regiao.loc[:,['atu_em_4_ano','tdi_em_4_ano','taxa_abandono_em_4_ano','tnr_em_4_ano']]

Index(['id_municipio', 'id_escola', 'localizacao', 'rede',
       'media_alunos_turma_em', 'media_alunos_turma_1', 'media_alunos_turma_2',
       'media_alunos_turma_3', 'afd_em_grupo_1', 'media_hora_em',
       'media_hora_1', 'media_hora_2', 'media_hora_3', 'distocao_idade_em',
       'distocao_idade_1', 'distocao_idade_2', 'distocao_idade_3',
       'taxa_abandono_em', 'taxa_abandono_em_1_ano', 'taxa_abandono_em_2_ano',
       'taxa_abandono_em_3_ano', 'prof_ensino_superior'],
      dtype='object')


KeyError: "None of [Index(['atu_em_4_ano', 'tdi_em_4_ano', 'taxa_abandono_em_4_ano',\n       'tnr_em_4_ano'],\n      dtype='object')] are in the [columns]"

A partir dessa análise, podemos ver que x das xx escolas 

Para ter uma visão geral do problema, calculei a mediana dos indíces para cada nível escolar: ensino básico, fundamental e médio.  

Primeiro analisei o banco de dados com os índices educacionais feito pelo INEP, que têm informações sobre a qualidade de ensino de cada escola, assim como dados sociais e econômicos dos alunos e professores. Os dados são por escola, então separei 

In [ ]:
pandas

[1] Municípios em dados
[2] https://www.ibge.gov.br/explica/codigos-dos-municipios.php#RJ
[3]http://www.gestaouniversitaria.com.br/artigos/conheca-os-indicadores-que-detalham-as-tendencias-da-educacao-basica-brasileira